test/shared_utilities/file_readers.jl

```diff
@@ -0,0 +1,64 @@
+#=
+    Unit tests for ClimaLand FileReaders module
+=#
+
+using Dates
+using Test
+
+import ClimaLand
+import ClimaLand: FileReaders
+using NCDatasets
+
+# Include testing artifacts from Bucket
+albedo_temporal_path = ClimaLand.Bucket.cesm2_albedo_dataset_path()
+albedo_bareground_path = ClimaLand.Bucket.bareground_albedo_dataset_path()
+include(joinpath(pkgdir(ClimaLand), "test", "artifacts", "artifacts.jl"))
+multiple_variables_path = era5_t2m_sp_u10n_dataset_path()
+
+@testset "NCFileReader with time" begin
+    PATH = albedo_temporal_path
+    NCDataset(PATH) do nc
+        available_dates = FileReaders.read_available_dates(nc)
+        @test length(available_dates) == 1980
+        @test available_dates[2] == DateTime(1850, 02, 14)
+
+        ncreader = FileReaders.NCFileReader(PATH, "sw_alb")
+
+        @test ncreader.dimensions[1] == nc["lon"][:]
+        @test ncreader.dimensions[2] == nc["lat"][:]
+
+        close(ncreader)
+
+        @test isempty(FileReaders.OPEN_NCFILES)
+    end
+
+    PATH = multiple_variables_path
+    NCDataset(PATH) do nc
+        ncreader_sp = FileReaders.NCFileReader(PATH, "sp")
+        ncreader_u = FileReaders.NCFileReader(PATH, "u10n")
+
+        # Test that the underlying dataset is the same
+        @test ncreader_u.dataset === ncreader_sp.dataset
+
+        # Test that we need to close all the variables to close the file
+        close(ncreader_sp)
+        @test !isempty(FileReaders.OPEN_NCFILES)
+        close(ncreader_u)
+        @test isempty(FileReaders.OPEN_NCFILES)
+    end
+end
+
+@testset "NCFileReader without time" begin
+    PATH = albedo_bareground_path
+    NCDataset(albedo_bareground_path) do nc
+        available_dates = FileReaders.read_available_dates(nc)
+        @test isempty(available_dates)
+
+        ncreader = FileReaders.NCFileReader(PATH, "sw_alb")
+
+        @test ncreader.dimensions[1] == nc["lon"][:]
+        @test ncreader.dimensions[2] == nc["lat"][:]
+
+        close(ncreader)
```

Comment:
we could check that `OPEN_NCFILES` is updated correctly in the case without time too

Reply:
It wouldn't test anything that we are not testing in the case with time